### PR TITLE
[WIP] Add entry point for e2e tests.

### PR DESCRIPTION
**Pull request by tcnghia**

### CODE REVIEW

File: third_party/istio-1.4-latest [hunk 1]
```diff
@@ -0,0 +1 @@
+istio-1.4.4/
```
Comment (mattmoor) [inline third_party/istio-1.4-latest]: Add trailing newline:
```suggestion
istio-1.4.4/

```

File: third_party/istio-1.4.4/download-istio.sh [hunk 2]
```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
```
Comment (chaodaiG) [inline third_party/istio-1.4.4/download-istio.sh]: nit, 2020 already

File: third_party/istio-1.4-latest (hunk 1 above)
Comment (mattmoor) [inline third_party/istio-1.4-latest]: Add trailing newline:
```suggestion
istio-1.4.4/

```

File: test/e2e-common.sh [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+  if ! which kail > /dev/null; then
+    bash <( curl -sfL https://raw.githubusercontent.com/boz/kail/master/godownloader.sh) -b "$GOPATH/bin"
+  fi
+
+
+  # Capture all logs.
+  kail > ${ARTIFACTS}/k8s.log.txt &
+  local kail_pid=$!
+  # Clean up kail so it doesn't interfere with job shutting down
+  add_trap "kill $kail_pid || true" EXIT
```
Comment (chaodaiG) [inline test/e2e-common.sh]: you'll have to define it here

Comment (chaodaiG) [reply test/e2e-common.sh]: https://github.com/knative/eventing/blob/master/test/e2e-common.sh#L56-L68

Comment (tcnghia) [reply test/e2e-common.sh]: Done, thanks!

File: test/e2e-tests.sh [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+source $(dirname $0)/e2e-common.sh
+
+
+# Script entry point.
+initialize $@  --skip-istio-addon
+
+go_test_e2e -timeout=20m -parallel=12 ./vendor/knative.dev/serving/test/conformance/ingress
```
Comment (chaodaiG) [inline test/e2e-tests.sh]:
```suggestion
go_test_e2e -timeout=20m -parallel=12 ./vendor/knative.dev/serving/test/conformance/ingress || fail_test
```

File: test/e2e-common.sh [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script includes common functions for testing setup and teardown.
+
+source $(dirname $0)/../vendor/knative.dev/test-infra/scripts/e2e-tests.sh
+
```
Comment (chaodaiG) [inline test/e2e-common.sh]:
```suggestion

function knative_setup() {
  start_latest_knative_serving
}

```

Comment (chaodaiG) [reply test/e2e-common.sh]: Not sure if this is what's desired, but just in case

File: test/e2e-tests.sh (hunk 4 above)
Comment (tcnghia) [reply test/e2e-tests.sh]: :man_facepalming: 

thanks :D

File: test/e2e-common.sh (hunk 5 above)
Comment (tcnghia) [reply test/e2e-common.sh]: I'm adding that, but it's slightly smaller than what Serving has.

File: third_party/istio-latest [hunk 6]
```diff
@@ -0,0 +1 @@
+istio-1.4.4
```
Comment (mattmoor) [inline third_party/istio-latest]: Add trailing newline:
```suggestion
istio-1.4.4

```

**Pull request closed without merging**